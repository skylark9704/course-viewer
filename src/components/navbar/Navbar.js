import React from "react";
import { NavLink } from "react-router-dom";

function Navbar(props) {
  const { items } = props;
  return (
    <div className="nav">
      {items.map((elem, index) => {
        return (
          <div key={index}>
            <NavLink
              exact={elem.exact}
              activeClassName="nav-active"
              to={elem.route}
            >
              {elem.label}
            </NavLink>{" "}
            {index !== items.length - 1 && "|"}
          </div>
        );
      })}
    </div>
  );
}

export default Navbar
