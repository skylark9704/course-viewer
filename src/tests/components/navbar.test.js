import React from "react";
import Navbar from "../../components/navbar";
import renderer from "react-test-renderer";

const tree = renderer.create(<Navbar items={[]} />).toJSON();

describe("Testing Navbar Component", () => {
  it("should render navbar", () => {
    expect(tree).toMatchSnapshot();
  });
});
