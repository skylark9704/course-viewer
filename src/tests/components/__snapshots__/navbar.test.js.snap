// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component should render navbar 1`] = `
<div
  className="nav"
/>
`;
